File: RockPaperScissors.java
import java.util.Random;
import java.util.Scanner;

public class RockPaperScissors {

	private String username;
	
	public static void main(String args[]) { 
		RockPaperScissors rockPaperScissors = new RockPaperScissors();
		System.out.println("Welcome to rock, paper, scissors!");
		rockPaperScissors.createUsername();
		String userPlay = rockPaperScissors.userPlay();
		String computerPlay = rockPaperScissors.computerPlay();
		
		int points = rockPaperScissors.playGame(userPlay, computerPlay);
		rockPaperScissors.results(points);
	}
	

	public int playGame(String userPlay, String computerPlay) {
		
		switch (userPlay) {
        case "r":
        	if (computerPlay.equals("s")) {
        		return 1;
        	} else if (computerPlay.equals("p")) {
        		return -1;
        	} else {
        		return 0;
        	}
        case "p":
        	if (computerPlay.equals("r")) {
        		return 1;
        	} else if (computerPlay.equals("s")) {
        		return -1;
        	} else {
        		return 0;
        	}
        case "s":
        	if (computerPlay.equals("p")) {
        		return 1;
        	} else if (computerPlay.equals("r")) {
        		return -1;
        	} else {
        		return 0;
        	}	
        }
		return 2;
	}
	
	private void results(int points) {
		if (points == 0) {
			System.out.println("Alas it was a draw");
		}else if (points == 1) {
			System.out.println("Whoop whoop you won " + username + "!");
		}else if (points == -1) {
			System.out.println("Boo you lost " + username + " :(");
		} else {
			System.out.println("There seems to have been a mistake somewhere");
		}
	}
	
	private String userPlay() {
		System.out.println("Please enter one of the following options:");
		System.out.println("R for rock, P for paper, S for scissors or E for exit");
		Scanner userInput = new Scanner(System.in);
		String userChoice = userInput.nextLine();
		
		if (userChoice.toUpperCase().equals("R")) {
			return "r";
		} else if (userChoice.toUpperCase().equals("P")) {
			return "p";
		} else if (userChoice.toUpperCase().equals("S")) {
			return "s";
		} else if (userChoice.toUpperCase().equals("E")) {
			System.out.println("Goodbye " + username + "...");
			System.exit(0);
			return "exited";
		} else {
			System.out.println("You didn't enter one of the options, I'm very dissapointed in you " + username);
			return userPlay();
		}
		
		
	}
	
	private String computerPlay() {
		Random random = new Random();
		int choice = random.nextInt(3) + 1;

		if (choice == 1) {
			System.out.println("The computer played rock");
			return "r";
		} else if (choice == 2) {
			System.out.println("The computer played paper");
			return "p";
		} else {
			System.out.println("The computer played scissors");
			return "s";
		}
	}

	public void createUsername() {
		System.out.println("Please enter a username :)");
		Scanner userInput = new Scanner(System.in); 
		this.username = userInput.nextLine();
	}
	
	

}

File: rockPaperScissors.cs
using System;
class RockPaperScissors {

  private string username;
  
  static void Main(string[] args)
  {
      Console.WriteLine("Welcome to rock, paper, scissors with C#\n");
      
      RockPaperScissors rockPaperScissors = new RockPaperScissors();
      rockPaperScissors.username = rockPaperScissors.createUsername();
      
      char userMove = rockPaperScissors.userPlay();

      char computerMove = rockPaperScissors.computerPlay();
      
      int points = rockPaperScissors.playGame(userMove, computerMove);
      rockPaperScissors.results(points);
  }
  
  
  int playGame(char userMove, char computerMove) {
	switch (userMove) {
	case 'r':
		if (computerMove == 's') {
			return 1;
		} else if (computerMove == 'p') {
			return -1;
		} else {
			return 0;
		}
	case 'p':
		if (computerMove == 'r') {
			return 1;
		} else if (computerMove == 's') {
			return -1;
		} else {
			return 0;
		}
	case 's':
		if (computerMove == 'p') {
			return 1;
		} else if (computerMove == 'r') {
			return -1;
		} else {
			return 0;
		}
	}
	return 2;
  }
  
  void results(int points) {
	if (points == 0) {
		Console.WriteLine("Alas it was a draw");
	} else if (points == 1) {
		Console.WriteLine("Whoop whoop you won " + username + "!");
	} else if (points == -1) {
		Console.WriteLine("Boo you lost " + username + " :(");
	} else {
		Console.WriteLine("There seems to have been a mistake somewhere");
	}
  }


  char userPlay() {
	
	Console.WriteLine("Please enter one of the following options:");
	Console.WriteLine("R for rock, P for paper, S for scissors and E for exit:  ");
	// substring here will only take the first character of the user
	// input so that this code can take a wider range of inputs
	// NEED TO LENGTH PROOF THIS 
	char userChoice = Convert.ToChar(Console.ReadLine().Substring(0, 1));
	
	if (char.ToUpper(userChoice) == 'R') {
		return 'r';
	} else if (char.ToUpper(userChoice) == 'P') {
		return 'p';
	} else if (char.ToUpper(userChoice) == 'S') {
		return 's';
	} else if (char.ToUpper(userChoice) == 'E') {
		Console.WriteLine("Goodbye...\n");
		Environment.Exit(0);
	} else {
		Console.WriteLine("you didn't enter any of the options provided, im very sad :(\n");
		return userPlay();
	}
	// this should never be reached?
	return 'n';
  }
  
  char computerPlay() {
	//this creates a random number between 1 and 3
	Random random = new Random();
	int randomNumber = random.Next(1, 4); 
	
	char compMove;
	if (randomNumber == 1) {
		Console.WriteLine("The computer played rock");
		compMove = 'r';
	} else if (randomNumber == 2) {
		Console.WriteLine("The computer played paper");
		compMove = 'p';
	} else if (randomNumber == 3) {
		Console.WriteLine("The computer played scissors");
		compMove = 's';
	} else {
		compMove = 'e';
		Console.WriteLine("error has occurred!\n");
	}
	return compMove;
  }

  
  string createUsername() {

	
	Console.WriteLine("Please enter your username:  ");
	string username = Console.ReadLine();
	
	return username;
  }
}
